endpoint.go

```diff
@@ -61,6 +61,7 @@ type endpoint struct {
 	generic       map[string]interface{}
 	joinLeaveDone chan struct{}
 	prefAddress   net.IP
+	prefAddressV6 net.IP
```

Comment:
+1 on keeping them separate because these are primary IP addresses of the container and primary IP addresses have a special meaning in that they have to be unique and so libnetwork needs to handle them that way and make appropriate IPAM calls to satisfy that requirement. In the future if we try to provide multiple IP addresses then we need to think about their characteristics and how libnetwork should handle them. But that can be kept separate from this discussion and PR.